common/tokenizer.cpp

```diff
@@ -85,18 +115,15 @@ U32String U32StringTokenizer::nextToken() {
 	// Skip delimiters when present at the beginning, to point to the next token
 	// For example, the below loop will set _tokenBegin & _tokenEnd to 'H' for the string -> "!!--=Hello World"
 	// And subsequently, skip all delimiters in the beginning of the next word.
-	while (_tokenBegin != _str.end() && _delimiters.contains(*_tokenBegin)) {
+	_tokenBegin = _tokenEnd;
```

Comment:
As you are changing this and getting closer to StringTokenizer, is there a reason why there are two different classes and not templates?
That would reduce code duplication.

Reply:
You'd have to ask the GSoC student who initially implemented this. I see that @sev- and @djwillis were the mentors for that task, perhaps they know why these classes are separate?

Reply:
no, no real reason. So, you may consider refactoring this too.

Reply:
I might do that at some point, but at the moment I don't know enough about how U32String works to do so... Are there any objections to merging this as-is?